mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala

```diff
@@ -58,9 +66,12 @@ class RankingMetrics[T: ClassTag](predictionAndLabels: RDD[(Array[T], Array[T])]
   @Since("1.2.0")
   def precisionAt(k: Int): Double = {
```

Comment:
Is there any similar notion of precision@k when the ground truth has a relevance 'weight'?

Reply:
I do not think so. As far as I know and search, there is no other definition of precision and the current implementation is enough. 

FYI: [Wikipedia for precision and recall](https://en.wikipedia.org/wiki/Precision_and_recall)